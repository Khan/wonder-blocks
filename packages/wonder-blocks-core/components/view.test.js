import React from "react";
import renderer from "react-test-renderer";

import View from "./view.js";

describe("View", () => {
    it("Should set the tag to be section", () => {
        // Arrage, Act
        const view = <View tag="section" />;
        const tree = renderer.create(view).toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Should set the tag to be section", () => {
        // Arrage, Act
        const view = <View tag="section" />;
        const tree = renderer.create(view).toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Should set the tag to be article", () => {
        // Arrage, Act
        const view = <View tag="article" />;
        const tree = renderer.create(view).toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Should set the tag to be aside", () => {
        // Arrage, Act
        const view = <View tag="aside" />;
        const tree = renderer.create(view).toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Should set the tag to be nav", () => {
        // Arrage, Act
        const view = <View tag="nav" />;
        const tree = renderer.create(view).toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Should set the tag to be div", () => {
        // Arrage, Act
        const view = <View />;
        const tree = renderer.create(view).toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });
});
