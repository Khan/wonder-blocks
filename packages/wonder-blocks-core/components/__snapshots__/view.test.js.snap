// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Should set the tag to be article 1`] = `
<article
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
/>
`;

exports[`View Should set the tag to be aside 1`] = `
<aside
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
/>
`;

exports[`View Should set the tag to be div 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
/>
`;

exports[`View Should set the tag to be nav 1`] = `
<nav
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
/>
`;

exports[`View Should set the tag to be section 1`] = `
<section
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
/>
`;

exports[`View Should set the tag to be section 2`] = `
<section
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
/>
`;
